bot/bot.go

```diff
@@ -94,6 +95,26 @@ func (b *Bot) Start() {
 	}
 }
 
+// Manages a Slack WebSocket connection and re-establishes the connection
+// when errors occur.
+func (b *Bot) manageConnection() {
+	for {
+		start := time.Now()
+		func() {
+			defer func() {
+				if r := recover(); r != nil {
+					b.log.Info("Recovered from panic in manageConnection")
+				}
+			}()
+			b.rtm.ManageConnection()
+		}()
+		if time.Since(start) < time.Second*3 {
+			// The connection is failing quite quickly. Let's back off.
+			time.Sleep(10)
```

Comment:
Should probably fail after a few consecutive retires.

Reply:
This is also wrong, `time.Sleep` takes `time.Duration` as an argument, not an `int`.